gem/lib/textmate_grammar/legacy_pattern.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require_relative 'pattern'
+
+class LegacyPattern < Pattern
```

Comment:
Will all of the Hash maps that get added to a grammar be converted to a LegacyPattern?
I see LegacyPattern imported into grammar.rb, but I don't see it used

Reply:
Currently it's just being included as the actual grammar uses it, the []= could detect if it's a legacy pattern hash and wrap with that instead, however.